cli/args/flags.rs

```diff
@@ -2582,8 +2593,6 @@ fn task_parse(
   }
 
   if let Some(mut index) = matches.index_of("task_name_and_args") {
```

Comment:
This index being out of sync with the raw_args index seems a bit error prone. In the interest of merging this for the release tomorrow, I'm going to push a change that keeps these in sync.

Reply:
Ok, I couldn't figure it out in a reasonable time. I moved this current code to be localized and it seems to work in a bunch of scenarios I tried so let's land it. Thanks!